api/prometheus/v1/api.go

```diff
@@ -231,9 +231,9 @@ type API interface {
 	// Flags returns the flag values that Prometheus was launched with.
 	Flags(ctx context.Context) (FlagsResult, error)
 	// LabelNames returns all the unique label names present in the block in sorted order.
-	LabelNames(ctx context.Context, startTime time.Time, endTime time.Time) ([]string, Warnings, error)
+	LabelNames(ctx context.Context, matches []string, startTime time.Time, endTime time.Time) ([]string, Warnings, error)
```

Comment:
This is technically a breaking change. I am not sure what are the rules for this project cc @beorn7 

Do we allow this or not in single major version?

Reply:
> In terms of tests we test what we have to test, no? We check if parameters are passed correctly to the underlying URL query.

Now only HTTP method and path are tested https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api_test.go#L71-L76. We don't check parameters.

Reply:
The whole API client is marked as experimental and not covered by the semantic versioning.

(In fact, I'd much prefer to have the API client on a separate release schedule. But as long as the API client is effectively pre-1.0, it doesn't really matter. Once we are ready to declare the API client as stable, we can revisit the decision if we want to keep it bundled with the rest of client_golang or separate it out (into a new repo or a separate Go module in a sub-directory).)